db/version_edit_handler.h

```diff
@@ -366,12 +366,13 @@ class DumpManifestHandler : public VersionEditHandler {
     // Write out each individual edit
     if (verbose_ && !json_) {
       // Print out DebugStrings. Can include non-terminating null characters.
-      fwrite(edit.DebugString(hex_).data(), sizeof(char),
-             edit.DebugString(hex_).size(), stdout);
+      std::string edit_dump_str = edit.DebugString(hex_);
+      fwrite(edit_dump_str.data(), sizeof(char), edit_dump_str.size(), stdout);
     } else if (json_) {
```

Comment:
nit: Not strictly part of the change in this PR, but since we are updating the area, I think the following would be slightly better
```
if (json_) {
...
} else if (verbose_) {
...
}
```